qualtricssurvey/models.py

```diff
@@ -663,29 +657,50 @@ def end_survey(self, data, suffix=''):  # pylint: disable=unused-argument
         }
         return response
 
+    def publish_grade(self):
+        grade_dict = {
+            'value': self.score.raw_earned,
+            'max_value': self.score.raw_possible,
+        }
+        LOGGER.info("Sanders", self.score)
+        self.runtime.publish(self, "grade", grade_dict)
+
     def has_submitted_answer(self):
         """
         Currently unused.
```

Comment:
@bsande6 Since we have to implement this function because we implement `ScorableXBlockMixin` and we would receive a `NotImplementedError` if we did not override this function per https://github.com/edx/XBlock/blob/master/xblock/scorable.py#L63-L67 my recommendation is to look into using an XBlock field like `is_answered` shown below and set this value to True whenever you set the `SurveyStatus.status = True`. 

It appears they call this `has_submitted_answer()` within the `ScorableXBlockMixin.rescore`. I would remove the `Currently unused.` comment and add that mention that this is called on a rescore from the `ScorableXBlockMixin parent class of the XBlock.

Whenever you set this field you can use it like so `if self.is_answered:`. This may clean up calls to the database and make your code a little easier to read. What are your thoughts on this changes?
```
    # Quick reference to see if learner has already answered the survey. This would alleviate any need to call the database everytime for SurveyStatus.status check.
    is_answered = Boolean(
        default=False,
        scope=Scope.user_state,
        help='Will be set to "True" if successfully answered'
    )
```